java/client/src/main/java/glide/managers/CommandManager.java

```diff
@@ -31,9 +32,26 @@ public <T> CompletableFuture<T> submitNewCommand(
         // when complete, convert the response to our expected type T using the given responseHandler
         return channel
                 .write(prepareRedisRequest(command.getRequestType(), command.getArguments()), true)
+                .exceptionally(this::exceptionHandler)
                 .thenApplyAsync(response -> responseHandler.apply(response));
     }
 
+    /**
+     * Exception handler for future pipeline.
+     *
+     * @param e An exception thrown in the pipeline before
+     * @return Nothing, it rethrows the exception
+     */
+    private Response exceptionHandler(Throwable e) {
+        if (e instanceof ClosingException) {
+            channel.close();
```

Comment:
Can this throw as well? Should the closing exception propagate or should it get logged?